### PR TITLE
Use device scale factor to support higher DPI on WPF

### PR DESCRIPTION
**Pull request by bjarteskogoy**
https://github.com/cefsharp/CefSharp/issues/836

### CODE REVIEW

**Comment by amaitland** [comment]
Is it possible to remove the `ScaleTransform` now?

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 1]
```diff
@@ -811,6 +812,12 @@ public BitmapInfo CreateBitmapInfo(bool isPopup)
             return BitmapFactory.CreateBitmap(isPopup);
         }
 
+        float _deviceScaleFactor = 1.0F;
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: Underscores only in `c++` code - consistency.
Variables should also be declared in the one place for consistency.

**Comment by amaitland** [comment]
Additional info on scaling in this thread https://code.google.com/p/chromiumembedded/issues/detail?id=946#c6

**Comment by bjarteskogoy** [comment]
Actually it's the combination of the ScaleTransform and device scale factor that makes it work properly. Without the transform and the device scale factor the wpf browser content would be the correct size on screen. WPF will upscale the image to comply with the current DPI setting, but this would just be an upscaling of the bitmap image -> very blurry. The scale transform keeps the pixel size from cef, which leads to a too small image when you have a higher DPI than 100%. The device scale factor "zooms" the rendered content from cef up to the correct size on screen.

**Comment by amaitland** [comment]
That's based on the transformed dimensions. I still think there's some investigation required.

**Comment by amaitland** [comment]
We should also expose this

http://magpcss.org/ceforum/apidocs3/projects/(default)/CefBrowserHost.html#NotifyScreenInfoChanged()

**Comment by bjarteskogoy** [comment]
The NotifyScreenInfoChanged should then probably be triggered in this method: https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Wpf/ChromiumWebBrowser.cs#L647

That would also mean that the scale transform needs to be updated at the same time, I believe.

**Comment by amaitland** [comment]
I've merged this though I still think further research is needed. The fact that it works is definitely an improvement, though I'm not convinced what we have is the `correct` solution.

> Currently, all sizes passed in and out of the API for both CEF and Chromium, are in view units (meaning virtual pixels or unscaled pixels) by design

 It's my understanding that `WPF` also uses device independent pixels and as such I'm wondering if the `ScaleTransform` is required because we provide transformed `Width/Height`. I don't have a `high dpi` device, so can't really toy with this easily.

**Pull request merged**